en/release-notes/upgrade-guide-v3.0.md

```diff
@@ -54,56 +48,166 @@ Please get the `UITransform` component on the node first, and then use the corre
+
+#### WeChat Mini Game
+
+The directory generated by Creator v2.4.3 after building the WeChat Mini Game is as follows:
+
+![image](wechat-v243.png)
+
+The directory generated by Creator 3.0 after building the WeChat Mini Game is as follows:
+
+![image](wechat-v3.png)
 
-The **compressed texture configuration** is modified to configure the preset in the project settings, then select the image resource in the resource manager, and then select the preset method. After the old project is upgraded, the editor will automatically scan all the compressed texture configurations in the project and sort out several presets. Since it is automatically scanned, the generated name may not be what you want, you can modify it here.
+From the above two figures, we can see that the directory generated after building the WeChat Mini Game, Creator v2.4.3 and 3.0 are mostly the same, with the following differences:
 
-![image](https://user-images.githubusercontent.com/1503156/100603295-d259f880-333f-11eb-8ff9-c985df953d83.png)
+1. Creator 3.0 puts all engine related code, such as core modules, physics modules, plugin scripts, etc., into the `wechatgame/cocos-js` directory. While Creator v2.4.3 scattered part of it in the `wechatgame` directory and part of it in the `wechatgame/cocos` directory.
 
-### TypeScript Reference Tutorial
+    ![image](wechat-cocosjs.png)
 
-- [TypeScript official website](https://www.typescriptlang.org/)
+2. Creator v2.4.3 compiles all the adaptation layer code of mini games into `adapter-min.js`, while Creator 3.0 stores all the adaptation layer code as loose files in the `libs` directory, without compilation.
+
+3. The startup script for Creator v2.4.3 is `main.js`, and for Creator 3.0 it is `application.js`.
+
+4. Creator v2.4.3 records all references of dynamic code in `ccRequire.js`. While Creator 3.0 does not have this feature currently.
+
+5. The `src/settings.js` used to manage configuration in Creator v2.4.3 is changed to `src/settings.json` in Creator 3.0.
+
+#### Native
+
+The directory generated by Creator v2.4.3 after building the Windows is as follows:
+
+![image](window-v243.png)
+
+The directory generated by Creator 3.0 after building the Windows is as follows:
+
+![image](window-v3.png)
+
+As you can see from the above two figures, there is a big difference between Creator v2.4.3 and Creator 3.0 in the directory generated after building the Windows platform.
+
+Since the underlying C++ code generated after building on each native platform (e.g. Android, Windows) is completely consistent. So in Creator 3.0, we extracted the underlying C++ code that was stored in the `jsb-link/frameworks/runtime-src/Classes` directory in Creator v2.4.3 and placed it in a shared `common-link` folder (or `common-default` folder, depending on the build template). This way, when building the native platform, if the `common-link` folder is detected to already exist, this part will no longer be processed, to speed up the build.
```

Comment:
So in Creator 3.0, we extracted the underlying C++ code
-->
**Cocos Creator v3.0**, extracts the underlying C++ code